tee-worker/litentry/core/stf-task/receiver/src/stf_task_receiver.rs

```diff
@@ -15,27 +15,30 @@
 // along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
 
 use crate::{
-	format, AuthorApi, Error, HandleState, Hash, SgxExternalitiesTrait, ShardIdentifier,
-	ShieldingCryptoDecrypt, ShieldingCryptoEncrypt, StfEnclaveSigning, StfTaskContext,
+	assertion::AssertionHandler, format, identity_verification::IdentityVerificationHandler,
+	AccessNodeMetadata, AuthorApi, CreateExtrinsics, Error, HandleState, Hash, IMPCallIndexes,
+	SgxExternalitiesTrait, ShardIdentifier, ShieldingCryptoDecrypt, ShieldingCryptoEncrypt,
+	StfEnclaveSigning, StfTaskContext, TaskHandler,
 };
 
-#[cfg(all(not(feature = "std"), feature = "sgx"))]
-use crate::chrono::{offset::Utc as TzUtc, TimeZone};
-
-#[cfg(feature = "std")]
-use chrono::{offset::Utc as TzUtc, TimeZone};
-
 use codec::Decode;
 use ita_sgx_runtime::IdentityManagement;
+use itp_node_api::metadata::pallet_vcmp::VCMPCallIndexes;
+use itp_ocall_api::EnclaveOnChainOCallApi;
 use lc_stf_task_sender::{stf_task_sender, RequestType};
-use litentry_primitives::{Assertion, Identity, Web2Network};
 use log::*;
-use std::string::String;
+use std::sync::Arc;
 
 // lifetime elision: StfTaskContext is guaranteed to outlive the fn
-pub fn run_stf_task_receiver<K, A, S, H>(context: &StfTaskContext<K, A, S, H>) -> Result<(), Error>
+pub fn run_stf_task_receiver<K, O, C, M, A, S, H>(
```

Comment:
I'm thinking a bit of restructuring might help here:
- merge this file to lib.rs -> `stf-task/receiver/src/stf_task_receiver.rs` is confusing
- maybe have a `handler` mod/folder and move the trait definition and implementation (AssertionBuildHandler, IdentityVerificationHandler) there

What do you think?

Reply:
I agree. It would look better.